### ISSUE TITLE
pipeline doesn't support objectMode

### ISSUE THREAD

**Issue by ronag** [mcollina/undici#221]
Currently it is not possible to return an object stream through client.pipeline.

e.g.

```js
client.pipeline(options, ({ body }) => {
  return pipeline(body, jsonParser, () => {}) // jsonParse is a string => object transform
})
```

Is this something that makes sense to support and if so how?

Should pipeline always be `readableObjectMode`, what are the downside of that? Or should pipeline have a `objectMode` parameter in `options`?

**Comment by ronag** [comment]
@mcollina @kmamal

**Comment by mcollina** [comment]
Is this a problem?

**Comment by kmamal** [comment]
I encountered this when requesting an rss feed and trying to pipe() the returned `body` to [feedparser](https://www.npmjs.com/package/feedparser) inside the handler. I was able to work around the limitation by moving the pipe() outside the handler.

That would have been a lot trickier to do if I needed to look at one of the headers to decide what to pipe to (for example piping to either an ndjson or xml parser based on content-type).
